webui/src/lib/components/repository/ChangeSummary.jsx

```diff
@@ -0,0 +1,108 @@
+import React, {useEffect, useState} from "react";
+import {
+    ClockIcon,
+    DiffAddedIcon,
+    DiffIgnoredIcon,
+    DiffModifiedIcon,
+    DiffRemovedIcon,
+    XCircleIcon
+} from "@primer/octicons-react";
+import {OverlayTrigger, Tooltip} from "react-bootstrap";
+import {humanSize} from "./tree";
+
+const MAX_NUM_OBJECTS = 10_000;
+const PAGE_SIZE = 1_000;
+
+class SummaryEntry {
+    constructor() {
+        this.count = 0
+        this.size = 0
+    }
+    add(count, size) {
+        this.count += count
+        this.size += size
+    }
+}
+
+class SummaryData {
+    constructor() {
+        this.added = new SummaryEntry()
+        this.changed = new SummaryEntry()
+        this.removed = new SummaryEntry()
+        this.conflict = new SummaryEntry()
+    }
+}
+
+/**
+ * Widget to display a summary of a change: the number of added/changed/deleted/conflicting objects.
+ * Shows an error if the change is has more than {@link MAX_NUM_OBJECTS} entries.
+
+ * @param {string} prefix - prefix to display summary for.
+ * @param {(after : string, path : string, useDelimiter :? boolean, amount :? number) => Promise<any> } getMore - function to use to get the change entries.
+ */
+export default ({prefix, getMore}) => {
+    const [resultsState, setResultsState] = useState({results: [], pagination: {}, tooManyPages: false});
+    const [loading, setLoading] = useState(true);
+    useEffect(async () => {
+        // get pages until reaching the max change size
+        if (resultsState.results && resultsState.results.length >= MAX_NUM_OBJECTS) {
+            setResultsState({results: null, pagination: {}, tooManyPages: true})
+            setLoading(false)
+            return
+        }
+        if (!loading) {
+            return
+        }
+        const {results, pagination} = await getMore(resultsState.pagination.next_offset || "", prefix, false, PAGE_SIZE)
+        if (!pagination.has_more) {
+            setLoading(false)
+        }
+        setResultsState({results: resultsState.results.concat(results), pagination: pagination, tooManyPages: false})
+    }, [resultsState.results, loading])
```

Comment:
This is inconsistent with file viewer. You may show statistics for 3 files but present 2 in the ui.
This happens because:
1. Listing happens separately.
2. Files are cached and statistics are not.

Reply:
Solving this basically means I have to refresh all expanded paths in the tree, together with all the stats that are already shown. 

Explanation:

Since summaries are calculated on demand - the only way for me to be consistent with the expanded tree is to refresh all expanded paths under the given path.

After doing so, shown summaries on upper levels should be updated to be consistent with their children. 
In turn, all expanded paths under these summaries should be updated to be consistent with them.

This potentially requires updating the whole tree.

I'm not sure it's worth the effort and the performance risk. Since it is clearly shown to the user that the summary is _calculated_, I think inconsistencies can be expected.

Reply:
@johnnyaug to clarify - are you referring to a situation where you added let's say 100 objects to partition "a", and at a time 50 objects are loaded to the tree but the stats show that 100 objects were added?

Reply:
What is described here is a scenario where data is constantly written to the repo, and you are viewing the change at different points in time.

Reply:
Just to make sure I understand, this limitation is due to the way we built the tree structure. "Refreshing" means we need to go and fetch everything again, while the summary already did it. I don't want to block on it because I don't think it would be that common. Can we open an issue for consolidation of the summary and the tree fetching?